py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,292 @@
+            longitudes.append(float(data[x]["station_gps_longitude"]))
+        self.gps_top_right_latitude = max(latitudes)
+        self.gps_top_right_longitude = max(longitudes)
+        self.gps_bot_left_latitude = min(latitudes)
+        self.gps_bot_left_longitude = min(longitudes)
+
+    def _get_velib_data(self):
+        url = "".join(
+            [
+                "https://www.velib-metropole.fr/webapi/map/details?",
+                "gpsTopLatitude=%s",
+                "&gpsTopLongitude=%s",
+                "&gpsBotLatitude=%s",
+                "&gpsBotLongitude=%s",
+                "&zoomLevel=%s",
+            ]
+        )
+        url = url % (
+            self.gps_top_right_latitude,
+            self.gps_top_right_longitude,
+            self.gps_bot_left_latitude,
+            self.gps_bot_left_longitude,
+            self.gps_zoom_level,
+        )
+
+        try:
+            return self.py3.request(url, timeout=self.request_timeout).json()
+        except self.py3.RequestException:
+            return None
+
+    def _manipulate(self, data):
+        index = 1
+        stations = {}
+
+        for code in self.station_codes:
+            new_station = {}
+
+            # search station
+            station = next(
+                (item for item in data if int(item["station"]["code"]) == int(code)),
+                None,
+            )
+
+            # flat, camel to snake and cast...
+            for k, v in self.py3.flatten_dict(station, delimiter="_").items():
+                new_station[self._camel_to_snake_case(k)] = self._cast_number(v)
+
+            # station_due_date_s: station due date in dateime iso format
+            # station_state_code: station code for thresholds
+            station_due_date = datetime.fromtimestamp(
+                int(new_station["station_due_date"])
+            )
+            new_station.update(
+                {
+                    "station_due_date_s": station_due_date.isoformat(),
+                    "station_state_code": int(
+                        self.station_states[new_station["station_state"]]
+                    ),
+                }
+            )
+
+            stations[index] = new_station
+            index += 1
+
+        return len(stations), stations
+
+    def velib_metropole(self):
+        if self.refresh is True:
+            # get number of stations and station list
+            json = self._get_velib_data()
+            if json:
+                self.number_of_stations, self.stations = self._manipulate(json)
+
+        # reset refresh
+        self.refresh = True
+
+        if not self.stations:
+            full_text = None
+        else:
+            # reset station_index counter
+            if self.station_index == 0:
+                self.station_index = 1
+
+            # thresholds
+            for x in self.thresholds:
+                if x in self.stations[self.station_index]:
```

Comment:
Your thresholds code looks off.

Reply:
Look at `loadavg`. You're missing `self.py3.get_color_names_list` as it's more useful than checking inside `thresholds` and allow things like `thresholds = [(0, 'bad'), (1, 'good')]` to work too.